internal/util-logging/src/main/scala/sbt/internal/util/ManagedLogger.scala

```diff
@@ -35,29 +35,53 @@ class ManagedLogger(
     )
   }
 
-  private lazy val SuccessEventTag = scala.reflect.runtime.universe.typeTag[SuccessEvent]
   // send special event for success since it's not a real log level
   override def success(message: => String): Unit = {
     if (terminal.fold(true)(_.isSuccessEnabled)) {
       infoEvent[SuccessEvent](SuccessEvent(message))(
         implicitly[JsonFormat[SuccessEvent]],
-        SuccessEventTag
+        StringTypeTag.default[SuccessEvent],
       )
     }
   }
 
-  def registerStringCodec[A: ShowLines: TypeTag]: Unit = {
+  @deprecated("Prefer version taking StringTypeTag", "1.4.0")
+  def registerStringCodec[A](
+      s: ShowLines[A],
+      tt: scala.reflect.runtime.universe.TypeTag[A]
+  ): Unit = {
+    LogExchange.registerStringCodec[A](s, tt)
+  }
+  def registerStringCodec[A: ShowLines: StringTypeTag]: Unit = {
     LogExchange.registerStringCodec[A]
   }
 
-  final def debugEvent[A: JsonFormat: TypeTag](event: => A): Unit = logEvent(Level.Debug, event)
-  final def infoEvent[A: JsonFormat: TypeTag](event: => A): Unit = logEvent(Level.Info, event)
-  final def warnEvent[A: JsonFormat: TypeTag](event: => A): Unit = logEvent(Level.Warn, event)
-  final def errorEvent[A: JsonFormat: TypeTag](event: => A): Unit = logEvent(Level.Error, event)
```

Comment:
These are public methods since `TaskStreams#log` returns `ManagedLogger`. My intent was to allow event logging from tasks.

Reply:
For logging it's probably safe to do Mima check even for `sbt.internal.*`.

Reply:
I left these public apis because zinc uses them so they can't be deleted until we publish a new util and update zinc. Having them around isn't that bad because they are awkward to call since I removed the implicits and forced the arguments to be called explicitly while preserving binary compatibility. Downstream users also generally wouldn't have to make source changes and will automatically pick up the new api on their next plugin release (or when they upgrade their build).

Reply:
The zinc usage is why I had to make the changes to BuildServerReporter below.